### PR TITLE
RELATED: RAIL-3383 - Sketch additional commands and events for the dashboard component

### PR DESCRIPTION
**Pull request by lupko**
---

Supported PR commands:

| Command                | Description            |
| ---------------------- | ---------------------- |
| `ok to test`           | Re-run standard checks |
| `extended test`        | BackstopJS tests       |
| `extended check sonar` | SonarQube tests        |

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `check-extended` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/3913/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5540/ : SUCCESS in 15m 24s

File: libs/sdk-ui-dashboard/src/model/commands/base.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+ * they target backend, workspace and dashboard in depending on the Dashboard component tree from which the dispatch
+ * is done.
+ *
+ * @internal
+ */
+export interface IDashboardCommand {
+    /**
+     * Command type. Always starts with "GDC.DASHBOARD.CMD"
+     */
+    readonly type: DashboardCommandType;
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/commands/base.ts:27]: It's a shame we do not have TypeScript 4.1 we could use template types for this.

File: libs/sdk-ui-dashboard/src/model/commands/dashboard.ts [hunk 2]
```diff
@@ -2,69 +2,165 @@
...
+/**
+ * Creates the LoadDashboard command. Dispatching this command will result in the load of all
+ * the essential data from the backend and initializing the state of Dashboard to a point where the
+ * dashboard can be rendered.
  *
+ * @param config - configuration to use for for the Dashboard; you MAY NOT provide any configuration or MAY provide
+ *  partial configuration. During the LoadDashboard processing the Dashboard component WILL resolve all the missing
+ *  parts by reading them from the backend
+ * @param permissions - permissions to use when determining whether the user is eligible for some actions with the
+ *  dashboard; you MAY NOT provide any permissions (undefined) and the Dashboard component WILL load the permissions
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/commands/dashboard.ts:29]: Nitpick: you MAY NOT souds like it is forbidden, which is not the case right? This is also used some lines above as well.

File: libs/sdk-ui-dashboard/src/model/events/dashboard.ts [hunk 3]
```diff
@@ -43,43 +17,33 @@ export interface IDashboardEvent {
...
          * Loaded dashboard.
          */
-        dashboard: IDashboard;
+        readonly dashboard: IDashboard;
 
         /**
          * Insights used on the dashboard.
          */
-        insights: IInsight[];
+        readonly insights: IInsight[];
```
Comment (no23reason) [inline libs/sdk-ui-dashboard/src/model/events/dashboard.ts:30]: Should the type be ReadonlyArray?

File: libs/sdk-ui-dashboard/src/model/commands/dashboard.ts (hunk 2 above)
Comment (lupko) [reply libs/sdk-ui-dashboard/src/model/commands/dashboard.ts]: reworded

**Comment by gdgate** [comment]
Build in pipeline check aborted.

File: libs/sdk-ui-dashboard/src/model/events/dashboard.ts (hunk 3 above)
Comment (lupko) [reply libs/sdk-ui-dashboard/src/model/events/dashboard.ts]: fixed

**Review by no23reason — APPROVED** [review]
LGTM

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul-el7/1145/ : SUCCESS in 1m 50s (non-voting)
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/1415/ : SUCCESS in 26s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/159/ : SUCCESS in 13m 28s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/5551/ : SUCCESS in 19m 07s

**Pull request merged**